src/kem/ntruprime/pqclean_sntrup761_avx2/params.h

```diff
@@ -32,7 +32,7 @@
 #define ppad 769
 #define endingmask _mm256_set_epi8(1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,0,0,0,0)
 #define crypto_core_weight PQCLEAN_SNTRUP761_AVX2_crypto_core_weightsntrup761
-#define p 761
+#define p_param 761
```

Comment:
I'm not immediately seeing the reason for this name change—could you explain why it's necessary?

Reply:
There is some conflict with openssl.
`In file included from /Volumes/GIT/git/oqs/liboqs/src/kem/ntruprime/pqclean_sntrup761_clean/kem.c:5:
In file included from /Volumes/GIT/git/oqs/liboqs/src/common/pqclean_shims/randombytes.h:6:
In file included from /Volumes/GIT/git/oqs/liboqs/build/include/oqs/rand.h:15:
In file included from /Volumes/GIT/git/oqs/liboqs/build/include/oqs/common.h:30:
In file included from /usr/local/opt/openssl@3/include/openssl/crypto.h:34:
In file included from /usr/local/opt/openssl@3/include/openssl/safestack.h:24:
/usr/local/opt/openssl@3/include/openssl/stack.h:45:60: error: expected ')'
void *OPENSSL_sk_delete_ptr(OPENSSL_STACK *st, const void *p);
                                                           ^
/Volumes/GIT/git/oqs/liboqs/src/kem/ntruprime/pqclean_sntrup761_clean/params.h:31:11: note: expanded from macro 'p'
#define p 761
          ^
/usr/local/opt/openssl@3/include/openssl/stack.h:45:28: note: to match this '('
void *OPENSSL_sk_delete_ptr(OPENSSL_STACK *st, const void *p);
                           ^
1 error generated.
[121/1636] Building C object src/kem/frodokem/CMakeFiles/frodokem.dir/external/frodo1344aes.c.o
ninja: build stopped: subcommand failed.`

Reply:
Hmmmm, I see. Is there a clean way to handle this with #include guards or something similar? Renaming seems like a brittle workaround. Suppose for argument's sake that a future version of OpenSSL changes the function parameter to `const void *q`—then everything would break here again.

Reply:
What I wouldn't give for C++ namespaces... :stuck_out_tongue_winking_eye:

Reply:
Resolved in https://github.com/open-quantum-safe/liboqs/pull/1926/commits/320c6e11d61f4c5c481b687f20f3999df411d3bb.